File: SecureUWPClient/SecureUWPClient/Ciphers/Symmetric/AES_CBC.cs
using SecureUWPClient.Ciphers.Symmetric;
using SecureUWPClient.Handshake;
using System;
using System.Diagnostics;
using System.Threading.Tasks;
using Windows.Security.Cryptography;
using Windows.Security.Cryptography.Core;
using Windows.Storage.Streams;

namespace SecureUWPClient.Ciphers.AES
{
    public class AES_CBC : Cryptography
    {
        private String encrypted;
        private String decrypted;
        private SupportedChipher cipher;
        private static byte[] ivBytes = new byte[]{0x15, 0x14, 0x13, 0x12, 0x11,
            0x10, 0x09, 0x08, 0x07, 0x06, 0x05, 0x04, 0x03, 0x02, 0x01, 0x00};
        public AES_CBC(SupportedChipher cipher) {
            this.cipher = cipher;
            this.encrypted = null;
            this.decrypted = null;
        }
        public async Task<String> AeS_Encrypt(string plainText, string EncryptionKey)
        {

            SymmetricKeyAlgorithmProvider SAP = SymmetricKeyAlgorithmProvider.OpenAlgorithm(SymmetricAlgorithmNames.AesCbcPkcs7);
            HashAlgorithmProvider HAP = HashAlgorithmProvider.OpenAlgorithm(HashAlgorithmNames.Md5);
            CryptographicHash Hash_AES = HAP.CreateHash();

            try
            {
                //byte[] KeyBytes = System.Text.Encoding.UTF8.GetBytes(password);
              //  byte[] KeyBytes16 = new byte[16];
                Hash_AES.Append(CryptographicBuffer.CreateFromByteArray(System.Text.Encoding.UTF8.GetBytes(EncryptionKey)));
                byte[] KeyBytes;
                CryptographicBuffer.CopyToByteArray(Hash_AES.GetValueAndReset(), out KeyBytes);
   
                CryptographicKey key = SAP.CreateSymmetricKey(CryptographicBuffer.CreateFromByteArray(KeyBytes));
                IBuffer Buffer = CryptographicBuffer.CreateFromByteArray(System.Text.Encoding.UTF8.GetBytes(plainText));
                IBuffer ivparams = CryptographicBuffer.CreateFromByteArray(ivBytes);
                encrypted = CryptographicBuffer.EncodeToBase64String(CryptographicEngine.Encrypt(key, Buffer, ivparams));

                return encrypted;
            }
            catch (Exception ex)
            {
                Debug.WriteLine(ex.ToString());
                return "Error in Encryption:With Aes ";
            }
        }

        public async Task<String> AeS_Decrypt(String EncryptedText, String DecryptionKey)
        {
            SymmetricKeyAlgorithmProvider SAP = SymmetricKeyAlgorithmProvider.OpenAlgorithm(SymmetricAlgorithmNames.AesCbcPkcs7);
            HashAlgorithmProvider HAP = HashAlgorithmProvider.OpenAlgorithm(HashAlgorithmNames.Md5);
            CryptographicHash Hash_AES = HAP.CreateHash();

            try
            {
                //byte[] KeyBytes = System.Text.Encoding.UTF8.GetBytes(password);
                Hash_AES.Append(CryptographicBuffer.CreateFromByteArray(System.Text.Encoding.UTF8.GetBytes(DecryptionKey)));
                byte[] KeyBytes;
                CryptographicBuffer.CopyToByteArray(Hash_AES.GetValueAndReset(), out KeyBytes);
              
                CryptographicKey key = SAP.CreateSymmetricKey(CryptographicBuffer.CreateFromByteArray(KeyBytes));


                IBuffer Buffer = CryptographicBuffer.DecodeFromBase64String(EncryptedText);
                byte[] Decrypted;

                IBuffer ivparams = CryptographicBuffer.CreateFromByteArray(ivBytes);
                CryptographicBuffer.CopyToByteArray(CryptographicEngine.Decrypt(key, Buffer, ivparams), out Decrypted);
                decrypted = System.Text.Encoding.UTF8.GetString(Decrypted, 0, Decrypted.Length);
                return decrypted;
            }
            catch (Exception ex)
            {
                Debug.WriteLine(ex.ToString());
                return "Error in Decryption:With Aes ";
            }

        }
    }
}

File: SecureAndroidClient/app/src/main/java/com/security/crypto/Ciphers/AES/AesCBC.java
package com.security.crypto.Ciphers.AES;


import com.security.crypto.Configuration.Properties;

import org.apache.commons.codec.binary.Base64;

import java.security.SecureRandom;
import java.security.Security;

import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;

public class AesCBC implements AesCiphers {
    private static byte[] ivBytes = new byte[]{0x15, 0x14, 0x13, 0x12, 0x11,
            0x10, 0x09, 0x08, 0x07, 0x06, 0x05, 0x04, 0x03, 0x02, 0x01, 0x00};
    private IvParameterSpec ivSpec;

    public AesCBC() {
        SecureRandom random = new SecureRandom();
        byte[] randBytes = new byte[16];
        random.nextBytes(randBytes);
        ivSpec = new IvParameterSpec(randBytes);
    }

    public String AeS_Encrypt(String plaintext, String key) throws Exception {
        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
        // convert plain text to bytes
        byte[] plainBytes = plaintext.getBytes(Properties.CHAR_ENCODING);

        Cipher cipher = Cipher.getInstance(Properties.AES_CBC);
        byte[] keybytes = Digest.Hash(key, Properties.MD5);
        SecretKeySpec sky = new SecretKeySpec(keybytes, Properties.AES_PROVIDER);
        cipher.init(Cipher.ENCRYPT_MODE, sky, new IvParameterSpec(ivBytes));
        // encrypt
        byte[] encrypted = cipher.doFinal(plainBytes);
        String encryptedString = new String(Base64.encodeBase64(encrypted));

        return encryptedString;

    }

    public String AeS_Decrypt(String encrypted, String key) throws Exception {
        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
        // convert plain text to bytes
        byte[] plainBytes = Base64.decodeBase64(encrypted.getBytes(Properties.CHAR_ENCODING));
        byte[] keybytes = Digest.Hash(key, Properties.MD5);
        SecretKeySpec sky = new SecretKeySpec(keybytes, Properties.AES_PROVIDER);
        Cipher cipher = Cipher.getInstance(Properties.AES_CBC);
        cipher.init(Cipher.DECRYPT_MODE, sky, new IvParameterSpec(ivBytes));

        byte[] decrypted = cipher.doFinal(plainBytes);
        return new String(decrypted, Properties.CHAR_ENCODING);
    }
}